File: 2047/main.cpp
#include <stdio.h>

// Array para armazenar o número de divisores para cada valor de 1 até 1586335
int ndivs[1586336], ans[100001];

int main() {
    int n, sum = 1586335;

    // Calcula o número de divisores para cada número de 1 até 1586335
    for (int i = 1; i <= sum; i++)
        for (int j = i; j <= sum; j += i)
            ndivs[j]++;

    // Calcula os primeiros 100000 números cujos divisores somam 1586335
    for (int i = 100000; i > 0; i--)
        ans[i] = ndivs[sum], sum -= ans[i];

    // Lê o valor de 'n' a ser impresso
    scanf("%d", &n);

    // Imprime os primeiros 'n' números cujos divisores somam 1586335
    for (int i = 1; i <= n; i++)
        printf("%d ", ans[i]);

    return 0;
}

File: 2047/main.py
def gera_sequencia(n):
    sequencia = []

    # Inicializa a sequência com os primeiros n números primos
    for i in range(1, n+1):
        sequencia.append(i)

    # Preenche o restante da sequência com números compostos
    for i in range(n+1, 2 * n + 1):
        sequencia.append(i)

    # Verifica a condição para a soma dos subconjuntos
    for k in range(2, n+1):
        soma = sum(sequencia[:k])
        divisores_soma = contar_divisores(soma)
        
        if divisores_soma != k:
            return "Impossível"

    return sequencia

def contar_divisores(numero):
    # Implemente a lógica para contar o número de divisores de 'numero'
    # Esta função depende de como você decide implementar o cálculo de divisores
    pass

# Leitura do valor de n
n = int(input())
resultado = gera_sequencia(n)

if resultado == "Impossível":
    print("Impossível")
else:
    print(" ".join(map(str, resultado)))
